pasha/context.py

```diff
@@ -39,37 +39,83 @@ def __init__(self, num_workers):
 
         self.num_workers = num_workers
 
-    def array(self, shape, dtype=np.float64):
+    @staticmethod
+    def _deprecated_array_function_warning():
+        from warnings import warn, filterwarnings
+        filterwarnings('always')
```

Comment:
This is probably a bit aggressive, a) because it's changing global state for the program, which is normally something libraries should avoid, and b) because setting warnings to always show means if a line inside a loop triggers a warning, it will be printed N times.

Deprecation warnings are a bit more visible than they used to be, but still not necessarily as visible if you'd want. If you want to [change the warning filter](https://docs.python.org/3/library/warnings.html#the-warnings-filter), you should set it to `'default'` specifically for for `DeprecationWarning`. Alternatively, people do sometimes abuse UserWarning to issue more visible deprecation warnings.

Reply:
You think that is aggressive, in my first iteration the functions were not even returning...

I'm with you and suspected you'd complain here. Quite frankly I was close to removing them altogether. There has not been a release yet, APIs may change. I'm hesitant to keep it quiet though.

How about making it a `print` and removing it in a bit? Just to make sure if someone stumbled on it by chance...

Reply:
Yeah, if you haven't done a release, I think you're pretty free to change the API without doing deprecations. Hopefully anyone using an unreleased project from Github is aware that it might change. Of course, it's a good idea to notify anyone you are aware of using it (e.g. David).